### DISCUSSION TITLE
Poly project create automation: specify project root and auto sync on project creation

### DISCUSSION THREAD

**Discussion by saifmode** [DavidVujic/python-polylith#287] [Ideas]
Hey, so I was playing around with polylith some more today. I had an idea, but first I wondered if this is a common use-case.

Firstly, I consider 'projects' to be my deployable units. For example, in my polylith I have a FastAPI base, and also a worker function component which I plan to have run on a daily schedule. So here, I have two potential projects, and each has a single 'root'. I'll call these 'root bricks'.

So my first question is, typically are 'projects' always focused around  such 'root bricks' (even though that root brick will pull functions in from other 'bricks')?

If so, then when it comes to creating projects, I think it would make sense to have an automation built around this.

Currently:

Whenever I create a project, I:
1. run `poly create project --name="<whatever>"`
2. add the 'root' brick to the list of bricks in that project's pyproject.toml
3. run `poly sync`, so all the other bricks are calculated and added to pyproject.toml

I think we could automate step 2 by adding an arg to the `poly create project` command. For example:
`uv run poly create project --root base/namespace/my_fast_api_app`
or
`uv run poly create project --root component/namespace/my_worker_function`

This would automatically add the following to the pyproject.toml `[tool.polylith.bricks]`, e.g.:

```
# my_fast_api_app project toml
"../../bases/name_space/my_fast_api_app" = "namespace/my_fast_api_app"
```
```
# my_worker_function project toml
"../../bases/name_space/my_worker_function" = "namespace/my_worker_function"
```

**Comment by DavidVujic** [comment]
The Polylith Architecture lets developers choose where to start: by coding and adding bricks, or by creating a project and setting up the infrastructure. I think the automation for step 2 would be cool thing for sure! Since the `create project` is a once-run thing and not a required workflow to do that in a certain order, I hesitate to add it to the tooling.

One goal I have is to keep the tool as simple as possible and the command options few (I have failed at that part in some of the commands). But there's always the option of adding custom/domain specific scripts (I think `uv` has a pretty good support support for this already), where you can combine the poly commands and bash or Python for automation (such as updating a pyroject.toml file).
